x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/lib/tracker.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import apm from 'elastic-apm-node';
+
+interface PdfTracker {
+  startLayout: () => void;
+  endLayout: () => void;
+  startScreenshots: () => void;
+  endScreenshots: () => void;
+  startSetup: () => void;
+  endSetup: () => void;
+  startAddImage: () => void;
+  endAddImage: () => void;
+  startCompile: () => void;
+  endCompile: () => void;
+  startGetBuffer: () => void;
+  endGetBuffer: () => void;
+  end: () => void;
+}
+
+const SPANTYPE_SETUP = 'setup';
+const SPANTYPE_OUTPUT = 'output';
+
+interface ApmSpan {
```

Comment:
@vigneshshanmugam I checked the elastic-apm-node library and didn't see any types exported. Did I miss something, or you know if it is possible to import the types?

If not, this is ok as it is, I guess. 😄

Reply:
Yeah you are right, not all interfaces and types are exported,  but I still don't know why you need only the `end` method here.

Reply:
APMSpan is the only type I need to reference, since I have them cached in this class. I only need to cache them so I can call `.end()` on them when needed.